src/Components/customEditor/customEditor.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component, Fragment } from 'react';
+import AceEditor from 'react-ace';
+import "ace-builds/src-min-noconflict/ext-language_tools";
+import "ace-builds/src-noconflict/mode-javascript";
+import "ace-builds/src-noconflict/snippets/javascript";
+import "ace-builds/src-noconflict/theme-monokai";
+import "ace-builds/src-noconflict/ext-emmet";
+import './editor.css';
+import Toolbar from '../toolbar/toolbar';
+import FileSaver from 'file-saver';
+
+export default class CustomEditor extends Component {
+
+    aceRef = null
+    
+    constructor() {
+        super()
+        this.state = {
+            isModalOpen: false,
+            fileName: '',
+            fileType: 'text/javascript'
+        }
+    }
+
+    componentDidMount() {
+        this.aceRef = this.refs.aceEditor;
+    }
+
+    onFileUpload = (file) => {
+        const fileName = file.name;
+        const fileType = file.type;
+        this.setState({fileName: fileName, fileType: fileType})
```

Comment:
I think you can also re-write this as.

`this.setState({ fileName, fileType });`